autonat.go

```diff
@@ -0,0 +1,142 @@
+package autonat
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+
+	host "github.com/libp2p/go-libp2p-host"
+	peer "github.com/libp2p/go-libp2p-peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+type NATStatus int
+
+const (
+	NATStatusUnknown = iota
```

Comment:
I would give those constants explicit type of `NATStatus`.

Reply:
sure, will do.